enhancements/helm3/console.md

```diff
@@ -0,0 +1,53 @@
+# Helm Charts in the Developer Catalog
+
+Helm is a Kubernetes package manager.  Helm 3.0 is a major release of helm which brings in a rich set of features and addresses major security concerns around tiller.  
+
+Red hat Openshift wants to bring Helm based content support to Openshift 4.4 Developer Catalog along with Operators to strengthen the helm based ecosystem.
+
+![Developer Catalog](../helm3/assets/dev-catalog.png)
+
+## Charts in the Developer Catalog
+
+The charts that would show up in the Developer Catalog will be powered by a [standard](https://helm.sh/docs/topics/chart_repository) helm charts' repository service.
+
+The 'data' in the service will be backed by a Github repository of helm charts.
+
+In the short-term, we would ship one helm charts' repository service *backed* by one public Github repository.
+
+## How would the UI discover the charts
+
+1. The UI would invoke `/api/console/helm/charts/index.yaml` to populate the available charts in the developer catalog. 
+
+2. This endpoint would proxy the request to the  charts' repository service, backed by a Github repository.
+
+In the initial phase, the charts repository service would be a served of a public Github repo.
+
+Example, 
+https://technosophos.github.io/tscharts/index.yaml is served directly off https://github.com/technosophos/tscharts . No in-cluster service will be used.
+
+The "Red Hat" public repository that would be used is https://github.com/redhat-developer/redhat-helm-charts ( work-in-progress )
+
+![Helm Charts Repo Service](../helm3/assets/charts-repo.png)
+
+## How would disconnected installs work 
+
+1. The user would need to 'clone' the public Github repository into her inside-the-network Github or Gitlab instance, and configure the same to serve static content ( "Pages" ).
+
+2. The URL serving the above static content would need to be configured in the cluster's Console Settings. The value would be propagated to the console backend through an environment variable.
```

Comment:
Configuring Helm repository location can be modelled similar to [`OperatorSource`](https://github.com/operator-framework/operator-marketplace/blob/7d230952a1045624b7601b4d6e1d45b3def4cf76/deploy/crds/operators_v1_operatorsource_crd.yaml). The resource needs to contain the following field:

```yaml
url: http://my.chart-repo.org/stable

# optional and only needed for UI purposes
displayName: myChartRepo

# optional and only needed for UI purposes
description: my private chart repo

# set to true if need to be disabled temporarly
disabled: true
```
The resource could be standalone, a part of [`Console` config](https://github.com/openshift/api/blob/master/config/v1/types_console.go#L26), or a part of [`Console` operator config](https://github.com/openshift/api/blob/master/operator/v1/types_console.go#L26)
* If standalone, its could be named as `HelmChartRepository` and be placed in (cluster-wide config)[https://github.com/openshift/api/tree/master/config/v1] or [console operator namespace](https://github.com/openshift/api/blob/master/operator/v1). The console operator would watch for changes on instances and reconfigure the console deployment.
* If a part of existing resource, it should be possible to specify a list of chart repositories, so that future use cases (e.g. [ODC-2994](https://issues.redhat.com/browse/ODC-2994)) could be implemented.

The console backend already implements a few helm endpoints (including the chart proxy), but the future might require to extract them into a separate service (e.g. to make openshift more modular). Hence, it would be good to define the configuration as the standalone resource (possibly cluster-wide or in a dedicated namespace, e.g. `openshift-helm`) detaching API from the implementation.